controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -222,5 +239,6 @@ public void close() {
         sealStreamLatency.close();
         updateStreamLatency.close();
         truncateStreamLatency.close();
+        INSTANCE.set(null);
```

Comment:
you are modifying a static variable in the close of an object reference. This may not be a good idea generally speaking. You will end up setting this static variable across close from multiple objects.

Reply:
Basically dont make this class Closable/AutoClosable. 
Dont let its users call close on it. 
They should call a new `static synchronized reset` method explicitly which takes the existing object and calls a private close on it. 
This will ensure there exists at maximum a single open object and there is never a closed object that getInstance method would return.

Reply:
Done.